File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

void dfs(int v, vector<vector<int>>& edge, vector<bool>& visited){
    visited[v] = true;
    cout << v << ' ';
    for(auto nxt : edge[v]){
        if(!visited[nxt]) dfs(nxt, edge, visited);
    }
}

void bfs(int v, vector<vector<int>>& edge, vector<bool>& visited){
    visited[v] = true;
    queue<int> q; q.push(v);
    while(q.size()){
        int cur = q.front(); q.pop();
        cout << cur << ' ';
        for(auto nxt : edge[cur]){
            if(!visited[nxt]){
                visited[nxt] = true;
                q.push(nxt);
            }
        }
    }
}

int main(){
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    int n, m, v; cin >> n >> m >> v;
    vector<vector<int>> edge(n+1);
    vector<bool> visited(n+1, false);
    for(int i = 0; i < m; i++){
        int x, y; cin >> x >> y;
        edge[x].push_back(y);
        edge[y].push_back(x);
    }
    for(int i = 1; i <= n; i++) sort(edge[i].begin(), edge[i].end());
    dfs(v, edge, visited);
    cout << '\n';
    fill(visited.begin(), visited.end(), false);
    bfs(v, edge, visited);
    return 0;
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
print = lambda *args, sep=" ", end="\n": sys.stdout.write(sep.join(map(str, args)) + end)
input = lambda: sys.stdin.readline().rstrip('\r\n')

from collections import deque

def main():
    n, m, v = map(int, input().split())
    edge = [[] for _ in range(n+1)]
    for _ in range(m):
        x, y = map(int, input().split())
        edge[x].append(y)
        edge[y].append(x)
    for e in edge: e.sort()
    visited = [False]*(n+1)
    def dfs(v):
        visited[v] = True
        print(v, end=' ')
        for nxt in edge[v]:
            if not visited[nxt]: dfs(nxt)
    dfs(v)
    print()
    visited = [False]*(n+1)
    def bfs(v):
        visited[v] = True
        dq = deque([v])
        while dq:
            cur = dq.popleft()
            print(cur, end=' ')
            for nxt in edge[cur]:
                if not visited[nxt]:
                    visited[nxt] = True
                    dq.append(nxt)
    bfs(v)
if __name__ == '__main__':
    sys.exit(main())